notmuch/mutt_notmuch.c

```diff
@@ -84,6 +85,31 @@ char *C_NmUnreadTag;  ///< Config: (notmuch) Tag to use for unread messages
 char *C_NmFlaggedTag; ///< Config: (notmuch) Tag to use for flagged messages
 char *C_NmRepliedTag; ///< Config: (notmuch) Tag to use for replied messages
 
+/**
+ * nm_hcache_open - Open a header cache
+ * @param m Mailbox
+ * @retval ptr Header cache handle
+ */
+static header_cache_t *nm_hcache_open(struct Mailbox *m)
+{
+#ifdef USE_HCACHE
```

Comment:
[1] I'm not a fan of defining an hcache-specific function then having it's behavior differ whether its been compile with or without a cache. It'll lead to some confusion...

Reply:
I agree - I think we should always compile hcache.c in, and let it return NULL on open and fetch and be a no-op on store. The rest of the code should be transparent. I think I can work on it on a separate issue.